### PR TITLE
Begin adding Windows Support

### PR DESCRIPTION
**Pull request by chreestopher**
Started to add windows support

### CODE REVIEW

File: lib/psaux.js [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+function parseWinProcesses(list, ps) {
+  list.push({
+    user: ps["User"],
+    pid: ps["ProcessID"],
+    cpu: parseFloat(ps["CPU"]),
+    mem: parseFloat(ps["Memory (MB)"]),
+    started: ps["StartTime"],
+    command: ps["CommandLine"],
+    name: ps["Name"]
+  });
```
Comment (zzarcon) [inline lib/psaux.js:43]: This are all the properties we can get from windows? will we nice if we can also add any of those `vsz`, `rss`, `tt`, `stat`.

File: lib/psaux.js [hunk 2]
```diff
@@ -1,6 +1,40 @@
 const execa = require('execa');
+const os = require('os');
 
-module.exports = options => {
+if (os.platform() != "win32") {
+  module.exports = getProcess
```
Comment (zzarcon) [inline lib/psaux.js]: Missing semicolon

File: lib/psaux.js [hunk 3]
```diff
@@ -1,6 +1,40 @@
...
+} else {
+  module.exports = getWinProcess
```
Comment (zzarcon) [inline lib/psaux.js]: Missing semicolon

File: lib/psaux.js [hunk 4]
```diff
@@ -1,6 +1,40 @@
...
+}
+
+function getWinProcess(options) {
+  return new Promise((resolve, reject) => {
+    execa('powershell', ['-noprofile',
+      '$perf = get-wmiobject -class Win32_PerfFormattedData_PerfProc_Process;$proc = get-wmiobject -class win32_process ;$TotalMem = get-wmiobject -class win32_computersystem | select -ExpandProperty TotalPhysicalMemory;$proc | foreach {$procID = $_;$procPerf = $perf | where {$_.creatingProcessID -eq $procID.ProcessID};$procID | select Name,@{n="User"; e= {$user = $_.getOwner(); if($user.user){"{0}\\{1}" -f $user.domain, $user.user}else {"System"}}},CommandLine,ProcessID,@{n="Memory (MB)";e={"{0:N3}" -f ($_.ws / 1mb)}},@{n="Memory (%)";e={"{0:N3}" -f ($_.ws / $TotalMem)}},@{n="StartTime";e={$_.converttoDateTime($_.CreationDate)}},@{n="CPU"; e={$item = $_;$COUNT = 0;if($procPerf.percentProcessorTime){$procPerf.PercentProcessorTime | foreach {$count += $_}; $percent = $count / $procPerf.PercentProcessorTime.length;"{0:N3}" -f [double]$percent} else {"{0:N3}" -f 0}}}} | ConvertTo-Json'
+    ]).then(result => {
+      var processes = JSON.parse(result.stdout);
```
Comment (zzarcon) [inline lib/psaux.js]: Could you add a `try catch` for check if the parsing fails? 

``` javascript
var processes;

try {
   processes = JSON.parse(result.stdout);
} catch(e) {
  processes = [];
}
```

**Comment by zzarcon** [comment]
@chreestopher thank you very much for the job! I just pointed some changes to make before get the PR merged. Please take a look and let me know what do you think ^^

File: lib/psaux.js (hunk 1 above)
Comment (chreestopher) [inline lib/psaux.js:43]: We can get more, but i wasnt sure exactly what the other properties
translate to in windows terms.

I will be busy tonight, but will try to implement the windows equivalents of vsz, rss, tt, stat.some time this weekend.

**Comment by zzarcon** [comment]
Good job @chreestopher, thanks for apply the changes. Will you be interested of being one of the maintainers of the repo? in that case i will add you as a member and we can build the next features we have in mind ;). Just let me know.

**Comment by chreestopher** [comment]
Sure! This was my first commit to a project other than my own, I would love
to continue to help out. My background is mostly in windows system
administration, so I can definitely help out in regards to the windows side
of things. Keep me posted!

**Pull request merged**